### PR TITLE
[EDA] Prevent Deletion of Account Record  if There are Associated Child Records

### LINKED ISSUE #1121: [EDA] Prevent Deletion of Account Record  if There are Associated Child Records
**Note: FOR INTERNAL USE ONLY. For support questions, please use the EDA Chatter group in [The Power of Us Hub](https://powerofus.force.com).
For feature requests, please use the Ideas tab in [The Power of Us Hub](https://powerofus.force.com).**

* **Do you want to request a *feature* or report a *bug*?**
Feature
* **What is the current behavior?**
If an Account has related child records, the Account record is deleted. 
* **What is the expected behavior?**
If an Account has related child records, the Account record should not be deleted. 
* **What is the motivation / use case for changing the behavior?**
data integrity

### PR DESCRIPTION
**Pull request by kdy618**
# Critical Changes

# Changes
## Updates to Ensure Data Integrity
We've expanded the functionality that prevents the deletion of certain records when they have related child records. Now, the following record types cannot be deleted if they have the following related records.

- Account records with related Address, Affiliation, Attribute (Agency record type), Course, Contact (Household and Business Org record type), Course Connection, Facility, Program Enrollment, Program Plan, Term, or Time Block records.
- Plan requirement records with related Plan Requirement records.
- Term records with related Term records.

This functionality is for internal use only and is enabled in Hierarchy Custom Settings for these objects (except Term) by default in new and existing orgs.  

If you've disabled any of the _object_ _CannotDelete_TDTM classes for these objects in an existing org, they will remain disabled even if they are enabled in Hierarchy Custom Settings.

**Note:** We strongly recommend that EDA Admins don't manage Hierarchy Custom Settings directly and, instead, use the EDA Settings UI, which will be provided in an upcoming release.

# Issues Closed
Closes #1121
# New Metadata

# Deleted Metadata

# Testing Notes
https://salesforce.quip.com/uDeQAAarQYHZ#EGQACAv00IY

### CODE REVIEW

File: src/classes/ACCT_CannotDelete_TEST.cls [hunk 1]
```diff
@@ -36,9 +36,834 @@
...
+        insert terms; 
+        
+        Course_Offering__c courseOffering1 = UTIL_UnitTestData_TEST.createCourseOffering(courses[0].Id, terms[0].Id); 
+        Course_Offering__c courseOffering2 = UTIL_UnitTestData_TEST.createCourseOffering(courses[1].Id, terms[1].Id); 
+        
+        Course_Enrollment__c courseConnection1 = UTIL_UnitTestData_TEST.getCourseConnection(newContacts[0].Id, courseOffering1.Id); 
+        Course_Enrollment__c courseConnection2 = UTIL_UnitTestData_TEST.getCourseConnection(newContacts[1].Id, courseOffering2.Id); 
+        courseConnection1.Account__c = newAccounts[0].Id;
+        courseConnection2.Account__c = newAccounts[1].Id;
+        insert new SObject[] {courseConnection1, courseConnection2};
```
Comment (MatthewBlanski) [inline src/classes/ACCT_CannotDelete_TEST.cls]: We seem to be using this insert new SObject[]{} format in some places which differs from the simple insert list format that we have in other pieces of code. Is there a reason we aren't using the same format?

Comment (kdy618) [reply src/classes/ACCT_CannotDelete_TEST.cls]: @MatthewBlanski , no good reason :( I'll update it to be consistent. 👍

**Comment by spinkelman** [comment]
Passes QA - Will merge once block is lifted.

**Comment by spinkelman** [comment]
Merging

**Pull request merged**